Add Subaru integration

G-Two opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
Adds documentation for new `subaru` integration https://github.com/home-assistant/core/pull/35760.


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/35760
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/1631
- This PR fixes or closes issue: N/A

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

G-Two committed:
Adds documentation for new subaru integration.

G-Two committed:
Update subaru.markdown

G-Two committed:
Corrected textlint errors.

frenck commented on source/_integrations/subaru.markdown:
```diff
@@ -0,0 +1,82 @@
+    
+    **Configuration** -> **Integrations** -> **Add** -> **Subaru**
+
+### YAML
```
Since this is a new integration, that interacts with a device or service, it should not supply YAML based configurations.

See ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md>

G-Two replied:
YAML references removed.  Thanks for pointing me towards the ADRs.

frenck commented on source/_integrations/subaru.markdown:
```diff
@@ -0,0 +1,82 @@
+
+This integration provides the following platforms:
+
+- Binary sensors - EV charger connection status.
+- Sensors - such as Outside temperature, odometer, estimated range, and EV information (battery level, range, charging rate).
+- Device tracker - obtain location of your car.
+- Lock - Lock and Unlock Doors.
+- Climate - Enables control of HVAC and Remote Engine Start (turn on/off, set target temperature).
+- Switch - Remote actuate Horn, Lights, request Location, and start EV charging.
```
```suggestion
- Binary sensors - EV charger connection status.
- Sensors - Such as Outside temperature, odometer, estimated range, and EV information (battery level, range, charging rate).
- Device tracker - Obtains the location of your car.
- Lock - Lock and Unlock Doors.
- Climate - Enables control of HVAC and Remote Engine Start (turn on/off, set target temperature).
- Switch - Remote actuate Horn, Lights, request Location, and start EV charging.
```

frenck reviewed (changes requested):
Thanks for the PR @G-Two!

I've reviewed it, and left some comments. Could you take a look? Thanks! 👍

G-Two committed:
Update source/_integrations/subaru.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

G-Two committed:
Corrections for #13493

- Remove YAML information
- Adds description of options
- Removes platforms omitted during initial PR (climate, device_tracker,
      binary_sensor, switch)
- Fixes typos

frenck committed:
:pencil2: Tweaks

frenck reviewed (dismissed):
✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

G-Two committed:
Updates due to changes in parent PR

- Only one platform for new integration (Lock removed)

bdraco commented on source/_integrations/subaru.markdown:
```diff
@@ -0,0 +1,45 @@
+ha_category:
+  - Car
+  - Sensor
+ha_release: 0.117
```
```suggestion
ha_release: 2021.3
```

bdraco commented on source/_integrations/subaru.markdown:
```diff
@@ -0,0 +1,45 @@
+
+- Sensors - Such as outside temperature, average fuel consumption, tire pressure, odometer, estimated range, and EV information (battery level, range, charging rate).
+
+## Configuration
+
+The Subaru integration offers configuration through the Home Assistant UI:
+    
+**Configuration** -> **Integrations** -> **Add** -> **Subaru**
+
+When prompted, enter the following configuration parameters:
+
+- **Email Address:** The email address associated with your MySubaru account.
+- **Password:** The password associated with your MySubaru account.
+- **PIN:** The PIN associated with your MySubaru account. 
+  - Note: If your account includes multiple vehicles, the same PIN will need to be used for all vehicles.
+
+First-time validation of your credentials may take up to 30 seconds. 
```
```suggestion
{% include integrations/config_flow.md %}
```

You'll need to rebase for that to work.

G-Two committed:
Update to align with parent PR

frenck reviewed (approved):
Thanks, @G-Two
